osu.Desktop/DiscordRichPresence.cs

```diff
@@ -115,6 +125,47 @@ private void updateStatus()
             client.SetPresence(presence);
         }
 
+        private void requestUserRulesetsRankings()
+        {
+            RulesetInfo cachedRuleset = ruleset.Value;
+
+            // When first logging in, we can't send a request with an empty id
```

Comment:
I don't understand. Can you explain why this is done this way? Ideally these two branches should not exist.

Reply:
That's my bad I made it because it was fired once at the very beginning when the user hadn't even logged in, but now that the api call is made only once the RPC has initialized, I can remove the part where user ID is not defined

Reply:
If one of the branches is no longer necessary then please remove.